File: MATTHEW/leetcode/AIcoding/5.Qwen-3/javascript_solutions/1318-minimum-flips-to-make-a-or-b-equal-to-c.js
function minFlips(a, b, c) {
    let flips = 0;
    while (a > 0 || b > 0 || c > 0) {
        const bitA = a & 1;
        const bitB = b & 1;
        const bitC = c & 1;
        if ((bitA | bitB) !== bitC) {
            if (bitC === 1) {
                flips += 1;
            } else {
                flips += (bitA + bitB);
            }
        }
        a >>= 1;
        b >>= 1;
        c >>= 1;
    }
    return flips;
}

File: MATTHEW/leetcode/AIcoding/4.GPT-4o/python3_solutions/1318-minimum-flips-to-make-a-or-b-equal-to-c.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        flips = 0
        for i in range(32):
            bitA = (a >> i) & 1
            bitB = (b >> i) & 1
            bitC = (c >> i) & 1

            if bitC == 1:
                if bitA == 0 and bitB == 0:
                    flips += 1
            else:
                flips += bitA + bitB
        return flips